File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> numSet = new HashSet<>();

        for (int n : nums) {
            numSet.add(n);
        }

        int maxLen = 0;
        for (int n : nums) {
            // Check if n is first num of consective sequence
            if (!numSet.contains(n - 1)) {
                int len = 1;
                while (numSet.contains(n + len)) {
                    len++;
                }
                maxLen = Math.max(maxLen, len);
            }
        }

        return maxLen;
    }
}

File: Data Structures & Algorithms/duplicate-integer/submission-1.cpp
#include <set>

class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
            std:set<int> tmp_set = {};
            for(const int& num: nums) {
                if (tmp_set.contains(num)) {
                    return true;
                }
                tmp_set.insert(num);
            }
            return false;
    }
};